sphinx/builders/latex.py

```diff
@@ -256,6 +258,23 @@ def copy_image_files(self):
                                    path.join(self.srcdir, src), err)
 
 
+class LaTeXPDFBuilder(LaTeXBuilder):
+    """
+    Utility builder that builds LaTeX output followed by a PDF.
+    """
+    name = 'latexpdf'
+
+    def finish(self):
+        # type: () -> None
+        super(LaTeXPDFBuilder, self).finish()
+        makecmd = os.environ.get('MAKE', 'make')
+        # TODO(stephenfin): Call things natively instead of using the Makefile
```

Comment:
the latex Makefile contains set-up for latexmk, if you move the logic here you will have to take into account setting of latex_engine.

and, the Japanese documents need to run extractbb on image files.

There is quite some logic built-in in the latex Makefile, and the reason we don't have a nice one for Windows is not only due to GNU/Make possibly lacking, it is also in syntax of system calls which are set-up in the latexmkrc and latexmkrcja files.

Hence, I am not sure what "natively" means, because it is always shell dependent in a way, in the end.

Reply:
Sorry, I meant using Python and `subprocess`, rather than the `make` wrapper. I'd like to investigate this further but it's difficult to do, as you say, and might not even be possible. As such, I've dropped the TODO and replaced it with a note explaining why we use the `make` command (basically what you just said)

Reply:
> There is quite some logic built-in in the latex Makefile, and the reason we don't have a nice one for Windows is not only due to GNU/Make possibly lacking, it is also in syntax of system calls which are set-up in the latexmkrc and latexmkjarc files.

Could you provide a little more info on this, just so I have the context? It seems that we don't use `latexmk` on Windows even though it's available. Is this because of those configuration files you're referring to above?

Reply:
@stephenfin sorry for delay. 

- First, a tool such as  ``Latexmk`` is needed: formerly Sphinx did 5 runs of ``pdflatex`` which in most cases is too much and in some cases possibly not enough (although circumstances are rare and possibly hard to recreate from a Sphinx project). At read the docs, last time I checked they do only 2 runs of ``pdflatex`` and this is not enough rtfd/readthedocs.org#2857; we needed to apply a trick at 6422f7b8c for our own docs, it is probable the problem is widespread across PDF builds at RTD. 

- Then ``Latexmk`` is a Perl script, which needs to be configured via a "latexmkrc" file. There we set up its Perl variables in particular for shell commands. Formerly all was in Makefile which miraculously worked on Windows. But at https://github.com/sphinx-doc/sphinx/blob/master/sphinx/texinputs/latexmkjarc#L3 in particular we need to workaround a bug of ``mendex`` executable which contrarily to ``makeindex`` raises error status if the ``foo.idx`` file generated from ``foo.tex`` is empty. This workaround was not needed when the rule was executed from Makefile. Now the Linux syntax of this is definitely not Windows-like, and for lack of expertise and time we had to postpone production of Windows-compatible syntax.

- Finally, many Windows users do not have a Perl available, and although [Latexmk](http://personal.psu.edu/jcc8/software/latexmk/) provides information how to install it, it looks as if activating its use by default for Sphinx builds will trigger many users into reporting issues (as #3807 which led to #3819).

For background on how Sphinx adopted Latexmk see #3082, but some tweaks were applied after the final version was merged.